### PR TITLE
anonymizer - add tests and split unit and integration

### PR DESCRIPTION
**Pull request by shiranr**
Add unit tests
move unit tests to correct place and add tests.

### CODE REVIEW

File: presidio-anonymizer/presidio_anonymizer/entities/analyzer_result.py [hunk 1]
```diff
@@ -78,15 +78,16 @@ def __str__(self):
         return f"start: {str(self.start)}, end: {str(self.end)}, " \
                f"score: {str(self.score)}, entity_type: {self.entity_type}"
 
-    def same_or_contained(self, other):
+    def has_conflict(self, other):
```
Comment (omri374) [inline presidio-anonymizer/presidio_anonymizer/entities/analyzer_result.py:81]: "conflicting" is a little confusing IMHO. Consider explaining what it means in the docstring or change the name.

File: presidio-anonymizer/tests/test_analyzer_result.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+         }, "Invalid input, analyzer result must contain score",),
+        ({
+             "start": 28,
+             "end": 32,
+             "score": 0.8,
+         }, "Invalid input, analyzer result must contain entity_type",),
+    ],
+    # fmt: on
+)
+def test_analyzer_result_fails_on_invalid_json_formats(request_json, result_text):
```
Comment (omri374) [inline presidio-anonymizer/tests/test_analyzer_result.py:181]: pytest has support for raised exceptions https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions.

File: presidio-anonymizer/tests/test_analyzer_results.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+from presidio_anonymizer.entities import AnalyzerResults, AnonymizerRequest
+
+
+def test_analyzer_results_not_failing_on_empty_list():
+    analyzer_result = AnalyzerResults()
+    assert len(analyzer_result.to_sorted_unique_results()) == 0
+
+
+def test_analyzer_results_sorted_set():
+    json_path = json_path = file_path("dup_payload.json")
```
Comment (omri374) [inline presidio-anonymizer/tests/test_analyzer_results.py:13]: consider turning into a fixture. Makes code a little more readable and would only require opening the file once for all tests.

File: presidio-anonymizer/tests/test_aonnymizer_request.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
+import json
```
Comment (omri374) [inline presidio-anonymizer/tests/test_aonnymizer_request.py:1]: typo in file name

File: presidio-anonymizer/tests/test_aonnymizer_request.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+                     "end": 32,
+                     "score": 0.8,
+                     "entity_type": "NUMBER"
+                 }
+             ]
+         }, "Invalid input, text can not be empty",)
+    ],
+    # fmt: on
+)
+def test_creating_anonymizer_request_should_fail_over_validation(request_json,
```
Comment (omri374) [inline presidio-anonymizer/tests/test_aonnymizer_request.py:58]: Cool!

File: presidio-anonymizer/tests/test_analyzer_result.py (hunk 2 above)
Comment (omri374) [reply presidio-anonymizer/tests/test_analyzer_result.py:181]: What would be the output if I'm missing more than one input field?

File: presidio-anonymizer/presidio_anonymizer/entities/analyzer_result.py (hunk 1 above)
Comment (shiranr) [reply presidio-anonymizer/presidio_anonymizer/entities/analyzer_result.py:81]: Will explain in doc string.

File: presidio-anonymizer/tests/test_analyzer_result.py (hunk 2 above)
Comment (shiranr) [reply presidio-anonymizer/tests/test_analyzer_result.py:181]: Will change to the supported raised exceptions.

File: presidio-anonymizer/tests/test_analyzer_results.py (hunk 3 above)
Comment (shiranr) [reply presidio-anonymizer/tests/test_analyzer_results.py:13]: Done.

File: presidio-anonymizer/tests/test_aonnymizer_request.py (hunk 4 above)
Comment (shiranr) [reply presidio-anonymizer/tests/test_aonnymizer_request.py:1]: Thanks :)

File: presidio-anonymizer/tests/test_analyzer_result.py [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+    [
+        (0, 10),
+        (10, 10),
+        (2, 8),
+        (0, 8),
+        (0, 10),
+    ],
+    # fmt: on
+)
+def test_analyzer_result_successfully_contains_another(start, end):
```
Comment (balteravishay) [inline presidio-anonymizer/tests/test_analyzer_result.py:18]: general comment about naming of test. should be more like:
whatIsTests_WithThisCondition_ExpectingThis
so something like:

contains_with_analyzer_result_is_success/true

**Review by omri374 — APPROVED** [review]

File: presidio-anonymizer/tests/test_analyzer_result.py (hunk 6 above)
Comment (shiranr) [reply presidio-anonymizer/tests/test_analyzer_result.py]: Done.

**Review by balteravishay — APPROVED** [review]

**Pull request merged**